animalcontrol.lua

```diff
@@ -0,0 +1,211 @@
+                                                --print("slaughter",state)
+                                                v.flags2.slaughter = state
+                                            end
+                                        end
+                                        attr = v.body.physical_attrs
+                                        if v.sex == 1 then
+                                            if args.showstats then
+                                                print(string.format("%-20s %-6s %-9d %-9d %-5s %-22s %-8s %-25s %-7d %-7d %-7d %-7d %-7d %-7d"
+                                                    ,name,v.id,k,v.race,sex
+                                                    ,tostring(v.flags2.slaughter),tostring(v.flags3.gelded),tostring(v.flags3.marked_for_gelding)
+                                                    ,attr.STRENGTH.value,attr.AGILITY.value,attr.TOUGHNESS.value,attr.ENDURANCE.value,attr.RECUPERATION.value,attr.DISEASE_RESISTANCE.value))
+                                            else
+                                                print(string.format("%-20s %-6s %-9d %-9d %-5s %-22s %-8s %-25s"
+                                                    ,name,v.id,k,v.race,sex
+                                                    ,tostring(v.flags2.slaughter),tostring(v.flags3.gelded),tostring(v.flags3.marked_for_gelding)))
+                                            end
+                                        else
+                                            if args.showstats then
+                                                print(string.format("%-20s %-6s %-9d %-9d %-5s %-22s %-8s %-25s %-7d %-7d %-7d %-7d %-7d %-7d"
+                                                    ,name,v.id,k,v.race,sex
+                                                    ,tostring(v.flags2.slaughter),"-","-"
+                                                    ,attr.STRENGTH.value,attr.AGILITY.value,attr.TOUGHNESS.value,attr.ENDURANCE.value,attr.RECUPERATION.value,attr.DISEASE_RESISTANCE.value))
+                                            else
+                                                print(string.format("%-20s %-6s %-9d %-9d %-5s %-22s %-8s %-25s"
+                                                    ,name,v.id,k,v.race,sex
+                                                    ,tostring(v.flags2.slaughter),"-","-"))
+                                            end
+                                        end
+                                    end
+                                end
+                            end
+                        end
+                    end
+                end
+            end
+        end
+    end
+    if not args.id then
+        if args.showstats then
+            print(string.format("%-20s %-6s %-9s %-9s %-5s %-22s %-8s %-25s %-7s %-7s %-7s %-7s %-7s %-7s"
+                , "animal type", "id", "unit id", "race id", "sex", "marked for slaughter", "gelded", "marked for gelding"
```

Comment:
It may be worth having this defined in just one place instead of repeating it here

Reply:
Probably. I am not sure if the strings are different in any way, but if they are the same you're right. I was likely just being lazy.